src/main/java/com/zufar/icedlatte/user/validator/PutUsersRequestValidator.java

```diff
@@ -0,0 +1,127 @@
+package com.zufar.icedlatte.user.validator;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.zufar.icedlatte.openapi.dto.AddressDto;
+import com.zufar.icedlatte.user.exception.PutUsersBadRequestException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class PutUsersRequestValidator {
+
+    private final int MIN_LENGTH = 2;
+    private final int MAX_LENGTH = 128;
+
+    public void validate(String firstName,
+                         String lastName,
+                         String phoneNumber,
+                         String birthDate,
+                         JsonObject addressDto) {
+        StringBuilder errorMessages = new StringBuilder();
+
+        StringBuilder firstNameParameterMessages = validateNameParameter(firstName, "First name");
+        errorMessages.append(firstNameParameterMessages);
+
+        StringBuilder secondNameParameterMessages = validateNameParameter(lastName, "Last name");
+        errorMessages.append(secondNameParameterMessages);
+
+        StringBuilder phoneNumberParameterMessages = validatePhoneParameter(phoneNumber);
+        errorMessages.append(phoneNumberParameterMessages);
```

Comment:
I think our current approach might not be the best way to handle this. It leans a lot on specific implementations and fields. If our requirements change down the line, we could end up having to expand this class significantly, which might make it tough to maintain. Plus, using reflection in classes that just handle Spring’s business logic seems like overkill.

Instead, how about we check out the `BindingResult` class from the `org.springframework.validation` package? It seems like a more straightforward solution for managing these types of issues, especially since they relate to the `MethodArgumentNotValidException`.

I'm suggesting we stick the javax.constraint validation annotations directly on the DTO class fields. We can put a `@Valid` annotation next to the `@RequestBody`. If something goes wrong, our centralized exception handler will catch it, pull all the errors from the Binding Result, pack them into a Map, and send it back to the client. Since we have auto-configuration, Jackson will handle turning that Map into JSON automatically, so we don’t have to mess with JsonObjects manually. What do you think?